src/metafile.c

```diff
@@ -1466,7 +1530,9 @@ GdipCreateMetafileFromFile (GDIPCONST WCHAR *file, GpMetafile **metafile)
 	
 	fp = fopen (file_name, "rb");
 	if (fp) {
-		status = gdip_get_metafile_from (fp, metafile, File);
+		if (gdip_get_metafile_from (fp, metafile, File) == Ok)
+			status = Ok;
```

Comment:
not sure why this change is required?

Reply:
`gdip_get_metafile_from` returns `OutOfMemory` for invalid metafiles, but `GdipCreateMetafileFromFile` returns `GenericError`
I added more tests to validate this - to be clearer, I also moved this to the 4th commit